framework/src/main/java/org/checkerframework/framework/flow/CFCFGBuilder.java

```diff
@@ -65,19 +77,33 @@ public static ControlFlowGraph build(
                         .process(root, underlyingAST);
         ControlFlowGraph phase2result = CFGTranslationPhaseTwo.process(phase1result);
         ControlFlowGraph phase3result = CFGTranslationPhaseThree.process(phase2result);
+        if (factory instanceof GenericAnnotatedTypeFactory) {
+            GenericAnnotatedTypeFactory<?, ?, ?, ?> asGATF =
+                    (GenericAnnotatedTypeFactory<?, ?, ?, ?>) factory;
+            asGATF.addSharedCFGForTree(underlyingAST.getCode(), phase3result);
+        }
         return phase3result;
     }
 
-    /*
-     * Given a SourceChecker and an AssertTree, returns whether the AssertTree
-     * uses an @AssumeAssertion string that is relevant to the SourceChecker.
+    /**
+     * Given a SourceChecker and an AssertTree, returns whether the AssertTree uses
+     * an @AssumeAssertion string that is relevant to the SourceChecker.
+     *
+     * @param checker the checker
+     * @param tree an assert tree
+     * @return true if the assert tree contains an @AssumeAssertion(checker) message string for any
+     *     subchecker of the given checker's ultimate parent checker
      */
     public static boolean assumeAssertionsActivatedForAssertTree(
-            SourceChecker checker, AssertTree tree) {
+            BaseTypeChecker checker, AssertTree tree) {
         ExpressionTree detail = tree.getDetail();
         if (detail != null) {
             String msg = detail.toString();
-            Collection<String> prefixes = checker.getSuppressWarningsPrefixes();
+            BaseTypeChecker ultimateParent = getUltimateParent(checker);
```

Comment:
The computation in the next 4 lines is not terribly expensive, but it is a touch expensive, and it occurs for every assertion with a detail message.  Could its value be cached in the checker?  (This would require adding a new method to the checker.)